pages/cookbook/nfts.mdx

```diff
@@ -2,8 +2,215 @@
+
+</Callout>
+
+Validation can be done using [`StateInit{:tact}`](/book/expressions#initof) and derived address of the NFT item.
+Initial data layout of the NFT item is shown below, but sometimes it may differ.
+
+```tact
+struct NFTItemInitData {
+    index: Int as uint64;
+    collectionAddress: Address;
+}
+
+inline fun calculateNFTAddress(index: Int, collectionAddress: Address, nftCode: Cell): Address {
+    let initData = NFTItemInitData{
+        index,
+        collectionAddress,
+    };
+
+    return contractAddress(StateInit{code: nftCode, data: initData.toCell()});
+}
+
+contract Example with Deployable {
+    nftCollectionAddress: Address;
+    nftItemIndex: Int as uint64;
+    nftCode: Cell;
+
+    init(nftCollectionAddress: Address, nftItemIndex: Int, nftCode: Cell) {
+        self.nftCollectionAddress = nftCollectionAddress;
+        self.nftItemIndex = nftItemIndex;
+        self.nftCode = nftCode;
+    }
+
+    receive(msg: NFTOwnershipAssigned) {
+        let expectedNftAddress = calculateNFTAddress(self.nftItemIndex, self.nftCollectionAddress, self.nftCode); // or you can even store expectedNftAddress
+        require(expectedNftAddress == sender(), "NFT contract is not the sender");
+
+        // your logic of processing nft ownership assign notification
+    }
+}
+```
+
+## Transferring NFT item
+
+To send jetton transfer use [`send(){:tact}`](/book/send) function.
+If you want to have full control over the sending process and require some complex logic for it,
+use [`nativeSendMessage(){:tact}`](/ref/core-advanced#nativesendmessage) function.
+
+```tact
+message(0x5fcc3d14) NFTTransfer {
+    queryId: Int as uint64;
+    newOwner: Address; // address of the new owner of the NFT item.
+    responseDestination: Address; // address where to send a response with confirmation of a successful transfer and the rest of the incoming message coins.
+    customPayload: Cell? = null; //  optional custom data. In most cases should be null
+    forwardAmount: Int as coins; // the amount of nanotons to be sent to the new owner.
+    forwardPayload: Slice as remaining; // optional custom data that should be sent to the new owner.
+}
+
+receive("transfer") {
+    let nftAddress = calculateNFTAddress(self.nftItemIndex, self.nftCollectionAddress, self.nftCode);
+
+    send(SendParameters{
+        to: nftAddress,
+        value: ton("0.1"),
+        body: NFTTransfer{
+            queryId: 42,
+            newOwner: address("NEW_OWNER_ADDRESS"), // NOTE: Modify it to your liking
+            responseDestination: myAddress(),
+            customPayload: null,
+            forwardAmount: 1,
+            forwardPayload: rawSlice("F"), // precomputed beginCell().storeUint(0xF, 4).endCell().beginParse()
+        }.toCell(),
+    });
+}
+```
+
+## Get NFT static info
+
+Note, that TON Blockchain does not allow contracts to call each other [getters](https://docs.tact-lang.org/book/contracts#getter-functions) like Ethereum does.
```

Comment:
Do we really need to mention Ethereum here?

Reply:
resolved